connector/connect/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -94,16 +94,16 @@ message Filter {
 message Join {
   Relation left = 1;
   Relation right = 2;
-  Expression on = 3;
-  JoinType how = 4;
+  Expression join_condition = 3;
+  JoinType join_type = 4;
 
   enum JoinType {
-    JOIN_TYPE_UNSPECIFIED = 0;
-    JOIN_TYPE_INNER = 1;
-    JOIN_TYPE_OUTER = 2;
-    JOIN_TYPE_LEFT_OUTER = 3;
-    JOIN_TYPE_RIGHT_OUTER = 4;
-    JOIN_TYPE_ANTI = 5;
+    JOIN_TYPE_INNER = 0;
```

Comment:
This will need an unspecified element - https://developers.google.com/protocol-buffers/docs/style#enums